main/integration-tests/support/resources/workflows/workflow-version.js

```diff
@@ -56,23 +55,35 @@ class WorkflowVersion extends Resource {
   // Triggers the workflow and pulls the the status of the workflow every second and only returns if the status is done
   // or until maxSecondsCount is reached. Default maxSecondsCount is 5 minutes.
   async triggerAndWait(body = {}, maxSecondsCount = 300) {
-    let counter = 0;
-    let status;
     const triggerInfo = await this.trigger(body);
     const instanceId = _.get(triggerInfo, 'instance.id');
+    await this.waitUntilComplete(instanceId, 1000, maxSecondsCount);
+    return triggerInfo;
+  }
 
+  // Polls a workflow at an intermittent interval until it has completed, failed, or it has reached the maxIntervalCount.
+  async waitUntilComplete(wfInstanceId, interval = 1000, maxIntervalCount = 300) {
```

Comment:
Do you think 5 minutes is enough for the workflow instance to complete? I've seen workspace provisioning take up to 8 minutes.

Reply:
That's the default params, in line 225 of create-workspace-service-catalog.js I set the interval as 10 seconds and the interval count to 48 -> 8 minutes total.